libr/asm/arch/arm/armass.c

```diff
@@ -421,6 +421,10 @@ static int thumb_assemble(ArmOpcode *ao, ut64 off, const char *str) {
 	} else
 	if (!strcmpnull (ao->op, "b") || !strcmpnull (ao->op, "b.n")) {
 		//uncond branch : PC += 4 + (delta*2)
+		int offset = getnum (ao->a[0]);
+		if (offset == 0) {
```

Comment:
True. I wrestled with this, because the `getnum` function uses `atoi` so there's no error condition. Also, -1 is technically also a valid number.

Does r2 codebase have a smarter way to accomplish this without needing RNum?